### PR TITLE
Add new CRD GCPTrafficDistributionPolicy.

### PR DESCRIPTION
**Pull request by cxhiano**

### CODE REVIEW

**Comment by gauravkghildiyal** [comment]
Can you please configure the CRD to receive the label as per https://gateway-api.sigs.k8s.io/geps/gep-2648/#standard-label-on-crd-objects

Refer https://github.com/GoogleCloudPlatform/gke-gateway-api/pull/13 for an example

**Comment by cxhiano** [comment]
> Can you please configure the CRD to receive the label as per https://gateway-api.sigs.k8s.io/geps/gep-2648/#standard-label-on-crd-objects
> 
> Refer #13 for an example

Added

**Comment by gauravkghildiyal** [comment]
Thanks!

**Review by briantkennedy — APPROVED** [review]

File: apis/networking/v1/shared_types.go [hunk 1]
```diff
@@ -53,3 +58,116 @@ const (
...
 	PolicyReasonTargetNotFound PolicyConditionReason = "TargetNotFound"
 )
+
+
+// LocalPolicyTargetReference identifies an API object to apply a direct or
+// inherited policy to. This should be used as part of Policy resources
+// that can target Gateway API resources. For more information on how this
+// policy attachment model works, and a sample Policy resource, refer to
+// the policy attachment documentation for Gateway API.
+type LocalPolicyTargetReference struct {
```
[the remark this answers is not in the document]
Comment (briantkennedy) [reply apis/networking/v1/shared_types.go:68]: nevermind, these aren't in v1 of v1.1

**Pull request merged**